mne/channels/channels.py

```diff
@@ -1007,11 +1011,129 @@ def _ch_neighbor_connectivity(ch_names, neighbors):
     ch_connectivity = np.eye(len(ch_names), dtype=bool)
     for ii, neigbs in enumerate(neighbors):
         ch_connectivity[ii, [ch_names.index(i) for i in neigbs]] = True
-
     ch_connectivity = sparse.csr_matrix(ch_connectivity)
     return ch_connectivity
 
 
+def find_ch_connectivity(info, ch_type):
+    """Find the connectivity matrix for the given data.
+
+    This function tries to infer the appropriate connectivity matrix template
+    for the given data. If a template is not found, the connectivity matrix is
+    computed using Delaunay triangulation based on 2d sensor locations.
+
+    Parameters
+    ----------
+    info : instance of mne.measuerment_info.Info
+        The measurement info.
+    ch_type : str
+        The channel type for computing the connectivity matrix. Currently
+        supports 'mag', 'grad' and 'eeg'.
```

Comment:
Coudl we also support `None`, which will throw an error if more than one channel type is present, but will work fine if only one is?